### PR TITLE
Converts/pgn converter

### PR DESCRIPTION
**Pull request by benrs**
PGN Converter using regex, we can implement strategies in the future so we can slowly phase this out if we think it's too slow or not intuitive.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,5 +2,6 @@ cmake_minimum_required(VERSION 3.9)
 project(stocklion)
 
 set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_BUILD_TYPE Debug)
```
Comment (benrs) [inline CMakeLists.txt]: Going to remove this, thought I needed it for debugging but my hosts file was causing issues.

File: util/UnicodeTable.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#include <iostream>
+#include <codecvt>
+#include "UnicodeTable.h"
+
+void UnicodeTable::printLine(std::string rowName, int length) {
```
Comment (benrs) [inline util/UnicodeTable.cpp:5]: This is the base for a unicode table for debugging, I'll finish it in another branch before next meet up.

File: converters/PGNConverter.cpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        std::string regString = std::regex_replace(regIterator->str(), newLine, " ");
+        std::smatch sm;
+        // Get the specific moves being made
+        std::regex_match(regString, sm, fullMoveRegEx);
+        std::string bothMoves = sm[1];
+
+        std::string whiteMove = bothMoves.substr(0, bothMoves.find(' '));
+        std::string blackMove = bothMoves.substr(bothMoves.find(' ') + 1, bothMoves.length());
+
+        playAlgebraicMove(WHITE, whiteMove, board);
```
Comment (jacobrs) [inline converters/PGNConverter.cpp]: playing 2 moves at a time is risky.

File: converters/PGNConverter.cpp [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            // Throw an exception since the pgn format is invalid
+            throw std::invalid_argument("The pgn passed in is invalid.");
+    }
+
+    if (!board->movePiece(startingPosition, endingPosition, currPlayer)) {
+        throw std::invalid_argument("The pgn passed in is invalid.");
+    }
+}
+
+int PGNConverter::getAlgebraicMoveType(std::string algebraicString) {
```
Comment (jacobrs) [inline converters/PGNConverter.cpp:74]: nice!

File: datastructures/Board.cpp [hunk 5]
```diff
@@ -142,4 +143,10 @@ void Board::printBoard() {
     }
 }
 
+Piece* Board::getPiece(int row, int col) {
```
Comment (jacobrs) [inline datastructures/Board.cpp]: Doesn't this encourage not using the Position datastructure?

File: main.cpp [hunk 6]
```diff
@@ -51,11 +53,18 @@ int main() {
...
-    FENConverter::convert("rnbqk2r/pppp1ppp/4pn2/1B1P4/1b6/4P3/PPP2PPP/RNBQK1NR w KQkq -", board);
+    Board *fenBoard = new Board(WHITE);
+    FENConverter::convert("rnbqk2r/pppp1ppp/4pn2/1B1P4/1b6/4P3/PPP2PPP/RNBQK1NR w KQkq -", fenBoard);
 
     std::cout << std::endl << "loaded from fen successfully" << std::endl;
-    board->printBoard();
+    fenBoard->printBoard();
+
+    Board *pgnBoard = new Board(WHITE);
+    PGNConverter::convert("1. d4 e5 2. dxe5 Qf6 3. Kd2 Qxe5 4. Nf3 Qe3+\n"
```
Comment (jacobrs) [inline main.cpp]: Maybe add an odd numbered test?

File: util/UnicodeTable.cpp (hunk 2 above)
Comment (jacobrs) [reply util/UnicodeTable.cpp:5]: This is gonna be nice

**Review by jacobrs — APPROVED** [review]

File: datastructures/Board.cpp (hunk 5 above)
Comment (benrs) [reply datastructures/Board.cpp]: Yeah, don't think I use it anyways, will remove it.

File: main.cpp [hunk 7]
```diff
@@ -6,6 +6,8 @@
 #include "datastructures/pieces/King.h"
 #include "datastructures/pieces/Knight.h"
 #include "converters/FENConverter.h"
+#include "datastructures/Board.h"
```
Comment (jacobrs) [inline main.cpp]: Do you use this include? It is marked as useless in clion.

**Pull request merged**